### PR TITLE
Make autosuggest trigger and options list less coupled

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

The autosuggest component is used internally by the property filter where the coming use case requires the autosuggest to be better abstract allowing for any content in the dropdown. The PR factors out a new internal element - autosuggest options list to reduce coupling between the internal input and list-options elements.

### How has this been tested?

Existing tests were updated to account for structural changes.

New tests added to cover the tab traps for recovery link.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [x] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autosuggest/__tests__/autosuggest.test.tsx [hunk 1]
```diff
@@ -6,6 +6,14 @@ import createWrapper from '../../../lib/components/test-utils/dom';
 import Autosuggest, { AutosuggestProps } from '../../../lib/components/autosuggest';
 import styles from '../../../lib/components/autosuggest/styles.css.js';
 import { expectNoAxeViolations } from '../../__a11y__/axe';
+import { KeyCode } from '@cloudscape-design/test-utils-core/dist/utils';
```
Comment (michaeldowseza) [inline src/autosuggest/__tests__/autosuggest.test.tsx:9]: `utils` is re-exported at the root so you should be able to import this with 

```
import { KeyCode } from '@cloudscape-design/test-utils-core/utils';
```

**Review by michaeldowseza — COMMENTED** [review]
Just want to clarify 1 part.

File: src/autosuggest/__tests__/autosuggest.test.tsx (hunk 1 above)
Comment (pan-kot) [reply src/autosuggest/__tests__/autosuggest.test.tsx:9]: Thanks, it was inserted by vs-code I think.

File: src/autosuggest/options-list.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const listRef = useRef<(index: number) => void>(null);
+
+  const highlightVisibleOption = useHighlightVisibleOption(options, setHighlightedIndex, isHighlightable);
+  const selectVisibleOption = useSelectVisibleOption(options, selectOption, isInteractive);
+  const handleMouseUp = createMouseEventHandler(selectVisibleOption, usingMouse);
+  const handleMouseMove = createMouseEventHandler(highlightVisibleOption, usingMouse);
+
+  useEffect(() => {
+    listRef.current?.(highlightedIndex);
+  }, [highlightedIndex]);
```
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/autosuggest/options-list.tsx:78]: Good question -- I will check.

**Review by michaeldowseza — APPROVED** [review]

**Pull request merged**